custom_components/elasticsearch/const.py

```diff
@@ -29,8 +36,12 @@
 
 VERSION_SUFFIX = "-v4_2"
 
```

Comment:
Introducing the new `valueas` field requires us to bump the index template version, so that the component knows to create a new template on startup. Without this change, ES refuses to ingest documents into existing indices because we do not allow dynamic fields. I'll include this in my PR

Reply:
Valueas shouldn't be getting populated when indexing directly without datastreams, is there more info on how you hit this?

Reply:
`valueas` is submitted as an empty object, which ES doesn't like. It's unclear if bumping the VERSION_SUFFIX actually helped, or if I imagined something. But if we shouldn't be sending this for legacy indexing, then it's probably more correct to omit the field entirely, rather than adjust the template.

```
2024-03-18 17:13:44.396 DEBUG (MainThread) [custom_components.elasticsearch] Collecting queued documents for publish
2024-03-18 17:13:44.396 INFO (MainThread) [custom_components.elasticsearch] Publishing 1 documents to Elasticsearch
2024-03-18 17:13:44.443 INFO (MainThread) [elasticsearch] POST https://my-es-instance:443/_bulk [status:200 request:0.045s]
2024-03-18 17:13:44.443 ERROR (MainThread) [custom_components.elasticsearch] Error publishing documents to Elasticsearch: ('1 document(s) failed to index.', [{'index': {'_index': 'hass-events-v4_2-000001', '_id': 'hqaOUo4BlVLLIvtAC7Uz', 'status': 400, 'error': {'type': 'strict_dynamic_mapping_exception', 'reason': '[1:513] mapping set to strict, dynamic introduction of [valueas] within [hass.entity] is not allowed'}, 'data': {'@timestamp': datetime.datetime(2024, 3, 18, 17, 13, 22, 432536, tzinfo=datetime.timezone.utc), 'hass.object_id': 'sun', 'hass.entity': {'id': 'sun.sun', 'domain': 'sun', 'attributes': {'next_dawn': '2024-03-19T05:10:12.556563+00:00', 'next_dusk': '2024-03-18T18:25:12.358264+00:00', 'next_midnight': '2024-03-18T23:48:02+00:00', 'next_noon': '2024-03-19T11:48:10+00:00', 'next_rising': '2024-03-19T05:44:44.577792+00:00', 'next_setting': '2024-03-18T17:50:34.066817+00:00', 'elevation': 5.03, 'azimuth': 262.73, 'rising': False, 'friendly_name': 'Sun'}, 'device': {}, 'value': 1, 'valueas': {}}, 'hass.domain': 'sun', 'hass.attributes': {'next_dawn': '2024-03-19T05:10:12.556563+00:00', 'next_dusk': '2024-03-18T18:25:12.358264+00:00', 'next_midnight': '2024-03-18T23:48:02+00:00', 'next_noon': '2024-03-19T11:48:10+00:00', 'next_rising': '2024-03-19T05:44:44.577792+00:00', 'next_setting': '2024-03-18T17:50:34.066817+00:00', 'elevation': 5.03, 'azimuth': 262.73, 'rising': False, 'friendly_name': 'Sun'}, 'hass.object_id_lower': 'sun', 'hass.entity_id': 'sun.sun', 'hass.entity_id_lower': 'sun.sun', 'hass.value': 1, 'agent.name': 'My Home Assistant', 'agent.type': 'hass', 'agent.version': '2023.11.1', 'ecs.version': '1.0.0', 'host.geo.location': {'lat': 52.3731339, 'lon': 4.8903147}, 'host.architecture': 'aarch64', 'host.os.name': 'Linux', 'host.hostname': 'c80a3e1a2441', 'tags': None}}}])
```

Reply:
fixed via commit